### PR TITLE
fix: auto menuPlacement to select and new styles

### PR DESCRIPTION
**Pull request by Butlerskyi**
Add prop to auto placement react-select menu

image

### CODE REVIEW

**Comment by Butlerskyi** [comment]
still need to fix some thing like arrow on input e.t.c, just want to know is this solution correct?

File: src/components/select/style.ts [hunk 1]
```diff
@@ -71,14 +77,18 @@ export const customStyles = (theme: PUITheme, error?: any, styles?: any): any =>
...
-      borderBottomColor: menuIsOpen && 'transparent',
+      borderBottomColor: placementState === 'top' ? null : menuIsOpen && 'transparent',
+      borderTopColor: placementState === 'top' ? menuIsOpen && 'transparent' : null,
       // borderBottomColor: menuIsOpen && 'transparent',
-      borderBottomLeftRadius: menuIsOpen && 0,
-      borderBottomRightRadius: menuIsOpen && 0,
+      borderBottomLeftRadius: placementState === 'top' ? null : menuIsOpen && 0,
+      borderTopLeftRadius: placementState === 'top' ? menuIsOpen && 0 : null,
+      borderBottomRightRadius: placementState === 'top' ? null : menuIsOpen && 0,
+      borderTopRightRadius: placementState === 'top' ? menuIsOpen && 0 : null,
```
Comment (vlacher12) [inline src/components/select/style.ts:86]: extact it to separate object pls

File: src/components/select/style.ts [hunk 2]
```diff
@@ -103,9 +115,12 @@ export const customStyles = (theme: PUITheme, error?: any, styles?: any): any =>
...
       border: `${isBorder()}`,
-      borderTop: menuIsOpen && 'none',
-      borderTopLeftRadius: menuIsOpen && 0,
-      borderTopRightRadius: menuIsOpen && 0,
+      borderTop: placementState === 'top' ? null : menuIsOpen && 'none',
+      borderBottom: placementState === 'top' ? menuIsOpen && 'none' : null,
+      borderTopLeftRadius: placementState === 'top' ? null : menuIsOpen && 0,
+      borderTopRightRadius: placementState === 'top' ? null : menuIsOpen && 0,
+      borderBottomLeftRadius: placementState === 'top' ? menuIsOpen && 0 : null,
+      borderBottomRightRadius: placementState === 'top' ? menuIsOpen && 0 : null,
```
Comment (vlacher12) [inline src/components/select/style.ts:123]: same here

File: src/components/select/select-component.tsx [hunk 3]
```diff
@@ -76,6 +76,12 @@ const Component = ({
     <CustomOption deleteItemIcon={deleteItemIcon} onDeleteItem={onDeleteItem} {...optionProps} />
   );
 
+  const [placementState, setPlacementState] = React.useState<string>('');
+
+  const getPlacement = (placement: string): void => {
+    setPlacementState(placement);
```
Comment (vlacher12) [inline src/components/select/select-component.tsx:82]: add check if state wasn't changed

**Review by vlacher12 — CHANGES_REQUESTED** [review]

**Comment by Butlerskyi** [comment]
[Missing margin (1.5d)](https://trello.com/c/1PG1isix/4626-missing-margin-15d)

**Review by vlacher12 — APPROVED** [review]

**Pull request merged**